modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/CefParser.java

```diff
@@ -0,0 +1,449 @@
+                        break;
+                    case 1:
+                        event.setDeviceVendor(headerFields.get(1));
+                        ingestDocument.setFieldValue("observer.vendor", headerFields.get(1));
+                        break;
+                    case 2:
+                        event.setDeviceProduct(headerFields.get(2));
+                        ingestDocument.setFieldValue("observer.product", headerFields.get(2));
+                        break;
+                    case 3:
+                        event.setDeviceVersion(headerFields.get(3));
+                        ingestDocument.setFieldValue("observer.version", headerFields.get(3));
+                        break;
+                    case 4:
+                        event.setDeviceEventClassId(headerFields.get(4));
+                        ingestDocument.setFieldValue("event.code", headerFields.get(4));
+                        break;
+                    case 5:
+                        event.setName(headerFields.get(5));
+                        break;
+                    case 6:
+                        event.setSeverity(headerFields.get(6));
+                        break;
+                }
+            }
+            String extensionString = cefString.substring(extensionStart);
+            Map<String, String> extensions = parseExtensions(extensionString);
+
+            if (removeEmptyValue) {
+                removeEmptyValue(extensions);
+            }
+            event.setExtensions(extensions);
+
+            // Translate possible ECS fields and remove them from extensions
+            Map<String, Object> translatedFields = extensions.entrySet()
+                .stream()
+                .filter(entry -> FIELD_MAPPING.containsKey(entry.getKey()))
+                .collect(Collectors.toMap(entry -> FIELD_MAPPING.get(entry.getKey()), entry -> {
+                    Class<?> fieldType = ECSFieldWithType.getFieldType(FIELD_MAPPING.get(entry.getKey()));
+                    return convertValueToType(entry.getValue(), fieldType);
+                }));
+
+            // Remove the translated entries from extensions
+            event.removeMappedExtensions();
+
+            ingestDocument.setFieldValue(targetField, event.toObject());
+            // Add ECS translations to the root of the document
+            if (translatedFields.isEmpty() == false) {
+                translatedFields.forEach(ingestDocument::setFieldValue);
+            }
+        } else {
+            throw new IllegalArgumentException("Invalid CEF format");
+        }
+    }
+
+    private Object convertValueToType(String value, Class<?> type) {
+        if (type == Long.class) {
+            return Long.parseLong(value);
+        } else if (type == Double.class) {
+            return Double.parseDouble(value);
+        } else if (type == Integer.class) {
+            return Integer.parseInt(value);
+        } else {
+            return value; // Default to String
```

Comment:
If the default is `String`, then I'm not sure I see the benefit of the many explicit String-typed mappings in `ECSFieldWithType`.

Reply:
I added 5c3ee2ba86155cd6bf6a5eb678f7c1603de3dcf4 and now I'm resolving this.